frontend/src/Editor/QueryManager/QueryManager.jsx

```diff
@@ -159,6 +159,9 @@ class QueryManagerComponent extends React.Component {
     //     }
     //   }
     // }
+    if (this.state.selectedQuery?.id !== nextProps.selectedQuery?.id) {
+      this.setState({ queryPreviewData: undefined });
+    }
```

Comment:
```suggestion
```

Instead of setting state in `componentWillReceiveProps`, set it in `setStateFromProps` where we set all the other states. It will avoid the one cycle of re-render. 

`queryPreviewData:
          this.state.selectedQuery?.id !== props.selectedQuery?.id ? undefined : this.state.queryPreviewData,`